doc/users.rst

```diff
@@ -1,602 +1,303 @@
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Foreshadow can optionally take in a :py:obj:`Preprocessor <foreshadow.preprocessor.Preprocessor>`
-object for the input data, a :py:obj:`Preprocessor <foreshadow.preprocessor.Preprocessor>` object for the target vector, a
-:py:obj:`sklearn.base.BaseEstimator` object to fit the preprocessed data, and a :py:obj:`sklearn.grid_search.BaseSearchCV`
-class to optimize the available hyperparameters.
+.. code-block:: python
 
-Here is an example of a fully defined :py:obj:`Foreshadow <foreshadow.foreshadow.Foreshadow>` object
+   predictions = shadow.predict(X_test)
 
 .. code-block:: python
 
-    shadow = fs.Foreshadow(X_preparer=Preprocessor(), y_preparer=Preprocessor(), estimator=AutoEstimator(), optimizer=None)
+   shadow.score(X_test, y_test)
 
-This code is equivalent to the :code:`fs.Foreshadow()` definition but explicitly defines each component. In order to disable one or more
-of these components simply pass :code:`False` to the named parameter (Note that the default :code:`None` automatically initializes the above).
+Model persistence
+=================
 
-:py:obj:`AutoEstimator <foreshadow.estimators.AutoEstimator>` is automatically defined as the estimator for Foreshadow. This estimator detects the problem type (classification or regression)
-and then either uses TPOT or Auto-Sklearn to serve as the estimator. The preprocessing methods are stripped from TPOT and Auto-Sklearn when they are used in this manner as we favor our own
-Preprocessor over their methods. As such these two frameworks will only perform model selection and estimator hyperparameter optimization by default.
+Save the fitted pipeline
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-**NOTE:** Future work includes implementing TPOT and AutoSklean's optimizers into this platform such that they can be used for both model selection and optimizing hyperparameters for the feature
-engineering aspects. Until then, however, they will only optimize the model as they are blind to the earlier parts of the pipeline.
-
-Foreshadow, acting as an estimator is also capable of being used in a :py:obj:`sklearn.pipeline.Pipeline` object. For example:
+After finding the best pipeline, you can export the fitted pipeline as a pickle file for your prediction task.
 
 .. code-block:: python
 
-    pipeline = Pipeline([("estimator", Foreshadow())])
-    pipeline.fit(X_train, y_train)
-    pipeline.score(X_test, y_test)
-
-By passing an optimizer into Foreshadow, it will attempt to optimize the pipeline it creates by extracting all the hyperparameters from
-the preprocessors and the estimator and passing them into the optimizer object along with the partially fit pipeline. This is a potentially
-long-running process and is not reccomended to be used with estimators such as TPOT or AutoSklearn which also do their own optimization.
-
+   pickled_fitted_pipeline_location = "fitted_pipeline.p"
```

Comment:
We should probably change the extension to `.pkl` just because it's what most people use :)

Reply:
Done.